hbase-client/src/main/java/org/apache/hadoop/hbase/client/ConnectionUtils.java

```diff
@@ -663,4 +664,13 @@ static void setCoprocessorError(RpcController controller, Throwable error) {
       controller.setFailed(error.toString());
     }
   }
+
+  static boolean isUnexpectedPreambleHeaderException(IOException e) {
+    if (!(e instanceof RemoteException)) {
+      return false;
+    }
+    RemoteException re = (RemoteException) e;
+    return FatalConnectionException.class.getName().equals(re.getClassName())
```

Comment:
Do we want a specific exception for this? Seems brittle to check message. I know we do that in other places, but just checking here in this new code

Reply:
I also want to check for a specific exception but actually the exception is thrown from old code...